internal/proxy/proxy.go

```diff
@@ -775,6 +767,58 @@ func newSocketMount(ctx context.Context, conf *Config, pc *portConfig, inst Inst
 	return m, nil
 }
 
+// newUnixSocketMount parses the configuration and returns the path to the unix
+// socket, or an error if that path is not valid.
+func newUnixSocketMount(inst InstanceConnConfig, unixSocketDir, version string) (string, error) {
+
+	var (
+		// the path to the unix socket
+		address string
+		// the parent directory of the unix socket
+		dir string
+	)
+
+	if inst.UnixSocketPath != "" {
+		// When UnixSocketPath is set
+		address = inst.UnixSocketPath
+
+		// If UnixSocketPath ends .s.PGSQL.5432, remove it for consistency
+		if strings.HasPrefix(version, "POSTGRES") && path.Base(address) == ".s.PGSQL.5432" {
```

Comment:
What happens if "address" doesn't exist?

Reply:
We return an error on line 803 if the parent directory for `address` doesn't exist.